packages/aws-cdk/lib/init.ts

```diff
@@ -76,27 +78,43 @@ export class InitTemplate {
                   + `(it supports: ${this.languages.map(l => colors.blue(l)).join(', ')})`);
             throw new Error(`Unsupported language: ${language}`);
         }
-        await this.installFiles(path.join(this.basePath, language), targetDirectory, {
+        const sourceDirectory = path.join(this.basePath, language);
+        await this.installFiles(sourceDirectory, targetDirectory, {
             name: decamelize(path.basename(path.resolve(targetDirectory)))
         });
+        await this.invokeHooks(sourceDirectory, targetDirectory);
     }
 
-    private async installFiles(srcDir: string, tgtDir: string, project: ProjectInfo) {
-        for (const file of await fs.readdir(srcDir)) {
-            const fromFile = path.join(srcDir, file);
-            const toFile = path.join(tgtDir, this.expand(file, project));
+    private async installFiles(sourceDirectory: string, targetDirectory: string, project: ProjectInfo) {
+        for (const file of await fs.readdir(sourceDirectory)) {
+            const fromFile = path.join(sourceDirectory, file);
+            const toFile = path.join(targetDirectory, this.expand(file, project));
             if ((await fs.stat(fromFile)).isDirectory()) {
                 await fs.mkdir(toFile);
                 await this.installFiles(fromFile, toFile, project);
                 continue;
             } else if (file.match(/^.*\.template\.[^.]+$/)) {
                 await this.installProcessed(fromFile, toFile.replace(/\.template(\.[^.]+)$/, '$1'), project);
+            } else if (file.match(/^.*\.hook\.[^.]+$/)) {
+                continue;
             } else {
                 await fs.copy(fromFile, toFile);
             }
         }
     }
 
+    private async invokeHooks(sourceDirectory: string, targetDirectory: string) {
```

Comment:
Add a comment explaining what this is doing

Reply:
✔️